src/helpers.rs

```diff
@@ -97,9 +97,18 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             Align::from_bytes(1).unwrap()
         )?.expect("we already checked for size 0");
 
-        let rng = this.memory_mut().extra.rng.get_mut();
         let mut data = vec![0; len];
-        rng.fill_bytes(&mut data);
+
+        if this.machine.communicate {
+            // Fill the buffer using the host's rng.
+            getrandom::getrandom(&mut data).map_err(|err| {
+                InterpError::Unsupported(UnsupportedOpInfo::Unsupported(err.to_string()))
```

Comment:
"Unsupported" does not seem right, does it?

Also we have `err_unsup!` so that you don't need to give these long paths.

Reply:
What error do you suggest?

Reply:
Well I guess "unsupported" is fine, but please don't just forward the error -- prefix something like "getrandom failed: " or so.

Reply:
Actually `err_unsup_format!` seems like the best fit here.

Reply:
I will fix this and expand the docs for the flag